engine/consensus/dkg/processor/engine.go

```diff
@@ -0,0 +1,193 @@
+package processor
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/dkg"
+	"github.com/onflow/flow-go/network"
+)
+
+// Engine implements the crypto DKGProcessor interface which provides means for
+// DKG nodes to exchange private and public messages. The same instance can be
+// used across epochs, but transitions should be accompanied by a call to
+// SetEpoch, such that DKGMessages can be prepended with the appropriate epoch
+// ID.
+type Engine struct {
+	unit      *engine.Unit
+	log       zerolog.Logger
+	me        module.Local
+	conduit   network.Conduit
+	msgCh     chan dkg.DKGMessage
+	committee flow.IdentifierList
+	myIndex   int
+	epochID   uint64
```

Comment:
```suggestion
	epochCounter   uint64
```

Another minor tweak to match terminology we use elsewhere